src/overlays/actors/ovl_En_Dragon/z_en_dragon.c

```diff
@@ -37,130 +72,777 @@ const ActorInit En_Dragon_InitVars = {
+        (fabsf(this->actor.world.pos.z - this->actor.home.pos.z) > 20.0f)) {
+        Math_ApproachF(&this->actor.world.pos.x, this->actor.home.pos.x, 0.3f, 300.0f);
+        Math_ApproachF(&this->actor.world.pos.z, this->actor.home.pos.z, 0.3f, 300.0f);
+        return;
+    }
+
+    sNumPythonsDead++;
+    if (sNumPythonsDead >= (BREG(39) + 8)) {
+        Math_Vec3f_Copy(&seahorsePos, &this->actor.parent->world.pos);
+        seahorsePos.x += (Math_SinS((this->actor.parent->world.rot.y + 0x8000)) * (500.0f + BREG(38)));
+        seahorsePos.y += -100.0f + BREG(33);
+        seahorsePos.z += (Math_CosS((this->actor.parent->world.rot.y + 0x8000)) * (500.0f + BREG(38)));
+        if (Actor_SpawnAsChildAndCutscene(&globalCtx->actorCtx, globalCtx, ACTOR_EN_OT, seahorsePos.x, seahorsePos.y,
+                                          seahorsePos.z, 0, this->actor.shape.rot.y, 0, 0x4000, this->actor.cutscene,
+                                          this->actor.unk20, NULL)) {
+            gSaveContext.save.weekEventReg[0xD] |= 1;
+            switch (this->pythonIndex) {
+                case 0:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x10;
+                    break;
+
+                case 1:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x20;
+                    break;
+
+                case 2:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x40;
+                    break;
+
+                case 3:
+                    gSaveContext.save.weekEventReg[0x53] |= 0x80;
+                    break;
+                case 4:
+                    gSaveContext.save.weekEventReg[0x54] |= 1;
+                    break;
+
+                case 5:
+                    gSaveContext.save.weekEventReg[0x54] |= 2;
+                    break;
+
+                case 6:
+                    gSaveContext.save.weekEventReg[0x54] |= 4;
+                    break;
+
+                case 7:
+                    gSaveContext.save.weekEventReg[0x54] |= 8;
+                    break;
+            }
+        }
+    }
+
+    Actor_MarkForDeath(&this->actor);
+}
+
+void EnDragon_UpdateDamage(EnDragon* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    u32 sp30;
+
+    if ((this->action == DEEP_PYTHON_ACTION_EXTEND) && ((this->collider.elements[2].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[3].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[4].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[5].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[6].info.bumperFlags & BUMP_HIT) ||
+                                                        (this->collider.elements[7].info.bumperFlags & BUMP_HIT))) {
```

Comment:
It doesn't seem like this actually matches? Which is silly because it's *logically* the same thing. So I guess we're stuck with this ugly-looking thing.

Reply:
Actually, I was just being silly, and this *does* match. I think it definitely looks better, so I'll go with that instead.